ADR/0044-spdx-support.md

```diff
@@ -0,0 +1,162 @@
+|                                 |     {                                                      |
+|                                 |          spdxElementId = `<ROOT-DOCUMENT-ID>`,             |
+|                                 |          relationshipType=DESCRIBES,                       |
+|                                 |          relatedSpdxElement=`<CONTAINER-IMAGE-ID>`,        |
+|                                 |     },                                                     |
+|                                 |     {                                                      |
+|                                 |          spdxElementId = `<A-BUILDER-IMAGE-ID>`,           |
+|                                 |          relationshipType=BUILD_TOOL_OF,                   |
+|                                 |          relatedSpdxElement=`<CONTAINER-IMAGE-ID>`         |
+|                                 |     }                                                      |
+|                                 | ]                                                          |
+
+*Explanation: Root document `DESCRIBES` `CONTAINER-IMAGE-ID` element which represents the container itself. `BUILDER-IMAGE-ID` represents the builder image which was used to build the container. The relationship type `BUILD_TOOL_OF` is used to express that the builder image was used to build the container image.*
+
+#### Metadata.tools
+The CycloneDX metadata.tools sub attributes that we are mostly interested in are the vendor and name elements. Information about the creation of the SPDX document can be stored into creationInfo. CreationInfo.creators element is basically a list of strings. There’s a vague specification ([here](https://spdx.github.io/spdx-spec/v2.3/document-creation-information/#68-creator-field)]) about how it should be structured in the standard. Strings should be formatted in the following way: `<Attribute>: <Value>`. For example vendor should be stored as `Vendor: <vendor>`
+
+| CyloneDX Attribute                             | SPDX Attribute                                    |
+|------------------------------------------------|---------------------------------------------------|
+| Metadata.tools = [{“vendor”: “X”, “name”: “Y”] | CreationInfo.creators = [“Vendor: X”, “Tool: Y”]  |
+
+#### Merging SPDX
+##### Packages
+Packages of two SPDX documents can be merged together as a concatenation of two lists. In cycloneDX component elements can have only a single purl attribute, therefore component elements representing packages with the same name and version but with different purl have to be stored as multiple elements. SPDX package elements can bear multiple purls. Therefore multiple cycloneDX components can be squashed together into single SPDX package element with purls concatenated into a single list. Following rules are applied to packages merging process:
+- Packages with the same purl's package name and version and type are squashed into single package element
```

Comment:
Similar concern to https://github.com/konflux-ci/architecture/pull/213#discussion_r1843490460 - this shouldn't be the authoritative descision on how to merge packages. Not all tools should do it this way, cachi2 certainly shouldn't

Reply:
Perhaps we can just re-word this to make it clear that this is a sort of default suggestions for how to do the merging if you don't have more specific requirements

Reply:
yeah, good point